gsa/src/web/pages/tlscertificates/detailspage.js

```diff
@@ -0,0 +1,295 @@
+ */
+import React from 'react';
+
+import _ from 'gmp/locale';
+
+import Divider from 'web/components/layout/divider';
+import HorizontalSep from 'web/components/layout/horizontalsep';
+import IconDivider from 'web/components/layout/icondivider';
+import Layout from 'web/components/layout/layout';
+
+import DetailsLink from 'web/components/link/detailslink';
+
+import ExportIcon from 'web/components/icon/exporticon';
+import ManualIcon from 'web/components/icon/manualicon';
+import ListIcon from 'web/components/icon/listicon';
+
+import Tab from 'web/components/tab/tab';
+import TabLayout from 'web/components/tab/tablayout';
+import TabList from 'web/components/tab/tablist';
+import TabPanel from 'web/components/tab/tabpanel';
+import TabPanels from 'web/components/tab/tabpanels';
+import Tabs from 'web/components/tab/tabs';
+
+import InfoTable from 'web/components/table/infotable';
+import TableBody from 'web/components/table/body';
+import TableData, {TableDataAlignTop} from 'web/components/table/data';
+import TableRow from 'web/components/table/row';
+
+import DetailsBlock from 'web/entity/block';
+import EntityPage, {Col} from 'web/entity/page';
+import EntityPermissions from 'web/entity/permissions';
+import {goto_list} from 'web/entity/component';
+import EntitiesTab from 'web/entity/tab';
+import EntityTags from 'web/entity/tags';
+import withEntityContainer, {
+  permissionsResourceFilter,
+} from 'web/entity/withEntityContainer';
+
+import DeleteIcon from 'web/entity/icon/deleteicon';
+import TlsCertificateIcon from 'web/components/icon/tlscertificateicon';
+
+import {selector, loadEntity} from 'web/store/entities/tlscertificates';
+
+import {
+  selector as permissionsSelector,
+  loadEntities as loadPermissions,
+} from 'web/store/entities/permissions';
+
+import PropTypes from 'web/utils/proptypes';
+
+import TlsCertificateDetails from './details';
+import TlsCertificateComponent from './component';
+
+const ToolBarIcons = ({
+  entity,
+  onTlsCertificateDeleteClick,
+  onTlsCertificateDownloadClick,
+}) => {
+  return (
+    <Divider margin="10px">
+      <IconDivider>
+        <ManualIcon
+          page="search"
+          searchTerm="tls certificates"
+          title={_('Help: TLS Certificate Assets')}
+        />
+        <ListIcon title={_('TLS Certificates List')} page="tlscertificates" />
+      </IconDivider>
+      <IconDivider>
+        <DeleteIcon entity={entity} onClick={onTlsCertificateDeleteClick} />
+        <ExportIcon
+          value={entity}
+          title={_('Export TLS Certififcate as XML')}
+          onClick={onTlsCertificateDownloadClick}
+        />
+      </IconDivider>
+    </Divider>
+  );
+};
+
+ToolBarIcons.propTypes = {
+  entity: PropTypes.model.isRequired,
+  onTlsCertificateDeleteClick: PropTypes.func.isRequired,
+  onTlsCertificateDownloadClick: PropTypes.func.isRequired,
+};
+
+const Details = ({entity, ...props}) => {
+  const showSourceBlock = [entity.sourceReportIds].reduce(
```

Comment:
I don't get this line :-) Any reason why not using `const showSourceBlock = entity.sourceReportIds.length > 0`?

Reply:
Yes, it was late at night ;) In the beginning I just did for the showXY what we do in the menubar. Turned out to be much simpler but I didn't remove the leftover...